Resource Acceptance Testing Missing CheckDestroy

## Description

[`TestCase`](https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCase) `CheckDestroy` functions are used to verify that test infrastructure has been removed at the end of an acceptance test (e.g. the `Delete` function of the resource works correctly) by calling the API to attempt to read the deleted infrastructure. More information can be found at: https://www.terraform.io/docs/extend/testing/acceptance-tests/testcase.html#checkdestroy

We would like to enable the [`tfproviderlint` `AT001` check](https://github.com/bflad/tfproviderlint/blob/master/passes/AT001/README.md) for verifying this check is included during pull request reviews, however there is some existing testing missing this functionality that needs to be addressed or explicitly ignored:

- [x] aws/provider_test.go:407:36: AT001: missing CheckDestroy
- [x] aws/provider_test.go:435:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_api_gateway_vpc_link_test.go:94:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_autoscaling_attachment_test.go:18:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_autoscaling_attachment_test.go:60:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_db_snapshot_test.go:17:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_db_snapshot_test.go:34:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_snapshot_copy_test.go:17:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_snapshot_copy_test.go:34:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_snapshot_copy_test.go:63:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_snapshot_copy_test.go:80:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:73:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:104:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:136:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:168:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:200:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:235:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:261:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ebs_volume_test.go:284:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_iam_instance_profile_test.go:44:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_iam_instance_profile_test.go:62:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_iam_instance_profile_test.go:78:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_inspector_resource_group_test.go:12:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_kms_ciphertext_test.go:10:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_kms_ciphertext_test.go:29:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_kms_ciphertext_test.go:48:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_lambda_event_source_mapping_test.go:271:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_lb_target_group_test.go:900:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_macie_member_account_association_test.go:37:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_ses_domain_dkim_test.go:21:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_snapshot_create_volume_permission_test.go:15:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_storagegateway_cache_test.go:75:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_storagegateway_cache_test.go:102:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_storagegateway_upload_buffer_test.go:75:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_storagegateway_working_storage_test.go:75:36: AT001: missing CheckDestroy
- [x] aws/resource_aws_vpc_dhcp_options_test.go:17:36: AT001: missing CheckDestroy

For a limited number of the above, bypassing the linter check either via `//lintignore:AT001` or `CheckDestroy: nil,` is acceptable as there is no API check to perform.

## Definition of Done

* `tfproviderlint` is updated to `v0.4.0` or later which will ignore data source files (upcoming release)
* In `GNUmakefile`, the `tfproviderlint` command under the `lint` target has the `-AT001` flag enabled and TravisCI testing passes

--- a/aws/resource_aws_kms_ciphertext_test.go
+++ b/aws/resource_aws_kms_ciphertext_test.go
@@ -8,8 +8,9 @@ import (
 
 func TestAccResourceAwsKmsCiphertext_basic(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
-		Providers: testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccResourceAwsKmsCiphertextConfig_basic,
@@ -27,8 +28,9 @@ func TestAccResourceAwsKmsCiphertext_validate(t *testing.T) {
 	resourceName := "aws_kms_ciphertext.foo"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
-		Providers: testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccResourceAwsKmsCiphertextConfig_validate,
@@ -46,8 +48,9 @@ func TestAccResourceAwsKmsCiphertext_validate_withContext(t *testing.T) {
 	resourceName := "aws_kms_ciphertext.foo"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
-		Providers: testAccProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccResourceAwsKmsCiphertextConfig_validate_withContext,

